src/transformStateWithClones.js

```diff
@@ -6,8 +6,30 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let newState = { ...state };
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      newState = { ...newState, ...obj.extraData };
+
+      result.push(newState);
+    } else if (obj.type === 'removeProperties') {
+      for (const key of obj.keysToRemove) {
+        const { [key]: removed, ...rest } = newState;
+
+        newState = rest;
+      }
+      result.push({ ...newState });
```

Comment:
You are repeating the action of pushing the newState to the result array in each condition. As per the checklist, perform this action just once after the switch block.